workflow-ui/backstack-android/src/main/java/com/squareup/workflow1/ui/backstack/StateRegistryHolder.kt

```diff
@@ -0,0 +1,148 @@
+package com.squareup.workflow1.ui.backstack
+
+import android.os.Bundle
+import androidx.lifecycle.Lifecycle.Event
+import androidx.lifecycle.Lifecycle.Event.ON_CREATE
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.savedstate.SavedStateRegistryController
+import androidx.savedstate.SavedStateRegistryOwner
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * TODO write documentation
+ */
+@OptIn(WorkflowUiExperimentalApi::class)
+internal class StateRegistryHolder(
```

Comment:
- [ ] This is a terrible name for this class, think of something better.

Reply:
`StateRegistryGroup`? `StateRegistryAggregator`? `ChildStateRegistryManager`?

Reply:
`Borg`

Reply:
`CompositeStateRegistryOwnerSupport`?

Reply:
+1 to `StateRegistryAggregator`. Succinct and captures the essentials.